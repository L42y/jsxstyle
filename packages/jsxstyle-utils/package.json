{
  "name": "jsxstyle-utils",
  "version": "2.0.6",
  "description": "Utilities used by jsxstyle and tooling built for jsxstyle",
  "author": "Pete Hunt",
  "license": "Apache-2.0",
  "main": "jsxstyle-utils.cjs.js",
  "module": "jsxstyle-utils.es.js",
  "jsnext:main": "jsxstyle-utils.es.js",
  "types": "jsxstyle-utils.d.ts",
  "keywords": [
    "jsxstyle"
  ]
}
