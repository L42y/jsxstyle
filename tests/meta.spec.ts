import { Dict } from 'jsxstyle-utils';
import * as PackageUtilities from 'lerna/lib/PackageUtilities';
import * as Repository from 'lerna/lib/Repository';
import * as packlist from 'npm-packlist';
import * as path from 'path';

interface Package {
  /** parsed contents of the package's package.json file */
  _package: {
    name: string;
    private?: true;
    version?: string;
    main?: string;
    scripts?: Dict<string>;
    devDependencies?: Dict<string>;
    dependencies?: Dict<string>;
  };
  /** absolute path to the package */
  _location: string;
}

// get all packages from Lerna
const JSXSTYLE_ROOT = path.resolve(__dirname, '..');
const repo = new Repository(JSXSTYLE_ROOT);
const packages: Package[] = PackageUtilities.getPackages(repo);
packages.sort((a, b) => a._package.name.localeCompare(b._package.name));

// separate private and public packages
const privatePackages = packages.filter(pkg => pkg._package.private);
const publicPackages = packages.filter(pkg => !pkg._package.private);

describe('jest', () => {
  it('picks the right packages', () => {
    expect.assertions(publicPackages.length + 1);
    let resolvedPaths = '\n';
    const maxPkgNameLen = Math.max(
      ...publicPackages.map(pkg => pkg._package.name.length)
    );

    publicPackages.forEach(pkg => {
      const { name } = pkg._package;
      const pkgPath = require.resolve(pkg._location);
      const resolvedPath = require.resolve(name);
      const relativePath = path.relative(JSXSTYLE_ROOT, pkgPath);
      const pad = ' '.repeat(maxPkgNameLen - name.length);
      resolvedPaths += `${name}${pad} --> ${relativePath}\n`;
      expect(pkgPath).toBe(resolvedPath);
    });

    expect(resolvedPaths).toMatchSnapshot();
  });
});

describe('npm publish', () => {
  it('only publishes the intended files', async () => {
    const privatePackageNames = privatePackages.map(pkg => pkg._package.name);
    expect(privatePackageNames).toMatchSnapshot('private packages');

    const pkgPromises = Promise.all(
      publicPackages.map(async pkg => {
        const fileList = await packlist({ path: pkg._location });
        return `
${pkg._package.name}
${pkg._package.name.replace(/./g, '=')}
${fileList.map(f => `- ${f}`).join('\n')}
`;
      })
    );

    await expect(pkgPromises).resolves.toMatchSnapshot('public packages');
  });
});
