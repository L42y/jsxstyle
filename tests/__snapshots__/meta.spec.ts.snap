// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest picks the right packages 1`] = `
"
jsxstyle        --> packages/jsxstyle/lib/jsxstyle.cjs.js
jsxstyle-loader --> packages/jsxstyle-loader/lib/loader.js
jsxstyle-utils  --> packages/jsxstyle-utils/lib/jsxstyle-utils.cjs.js
"
`;

exports[`npm publish only publishes the intended files 1`] = `
Array [
  "
jsxstyle
========
- package.json
- LICENSE
- README.md
- lib/jsxstyle.cjs.js
- lib/jsxstyle.d.ts
- lib/jsxstyle.es.js
- lib/jsxstyle.js
- preact/lib/jsxstyle-preact.cjs.js
- preact/lib/jsxstyle-preact.d.ts
- preact/lib/jsxstyle-preact.es.js
- preact/lib/jsxstyle-preact.js
- preact/package.json
",
  "
jsxstyle-loader
===============
- package.json
- LICENSE
- README.md
- lib/loader.d.ts
- lib/loader.js
- lib/loader.js.map
- lib/plugin.d.ts
- lib/plugin.js
- lib/plugin.js.map
- lib/types.d.ts
- lib/types.js
- lib/types.js.map
- lib/utils/ast/accessSafe.d.ts
- lib/utils/ast/accessSafe.js
- lib/utils/ast/accessSafe.js.map
- lib/utils/ast/evaluateAstNode.d.ts
- lib/utils/ast/evaluateAstNode.js
- lib/utils/ast/evaluateAstNode.js.map
- lib/utils/ast/extractStaticTernaries.d.ts
- lib/utils/ast/extractStaticTernaries.js
- lib/utils/ast/extractStaticTernaries.js.map
- lib/utils/ast/extractStyles.d.ts
- lib/utils/ast/extractStyles.js
- lib/utils/ast/extractStyles.js.map
- lib/utils/ast/generatedUid.d.ts
- lib/utils/ast/generatedUid.js
- lib/utils/ast/generatedUid.js.map
- lib/utils/ast/getPropValueFromAttributes.d.ts
- lib/utils/ast/getPropValueFromAttributes.js
- lib/utils/ast/getPropValueFromAttributes.js.map
- lib/utils/ast/getSourceModule.d.ts
- lib/utils/ast/getSourceModule.js
- lib/utils/ast/getSourceModule.js.map
- lib/utils/ast/getStaticBindingsForScope.d.ts
- lib/utils/ast/getStaticBindingsForScope.js
- lib/utils/ast/getStaticBindingsForScope.js.map
- lib/utils/ast/parse.d.ts
- lib/utils/ast/parse.js
- lib/utils/ast/parse.js.map
- lib/utils/getClassNameFromCache.d.ts
- lib/utils/getClassNameFromCache.js
- lib/utils/getClassNameFromCache.js.map
- lib/utils/getStylesByClassName.d.ts
- lib/utils/getStylesByClassName.js
- lib/utils/getStylesByClassName.js.map
- lib/utils/wrapFileSystem.d.ts
- lib/utils/wrapFileSystem.js
- lib/utils/wrapFileSystem.js.map
- plugin/package.json
- schema/loader.json
",
  "
jsxstyle-utils
==============
- package.json
- LICENSE
- README.md
- lib/addStyleToHead.d.ts
- lib/componentStyles.d.ts
- lib/dangerousStyleValue.d.ts
- lib/getStyleCache.d.ts
- lib/getStyleKeysForProps.d.ts
- lib/hyphenateStyleName.d.ts
- lib/jsxstyle-utils.cjs.js
- lib/jsxstyle-utils.d.ts
- lib/jsxstyle-utils.es.js
- lib/jsxstyle-utils.js
- lib/stringHash.d.ts
- lib/types.d.ts
",
]
`;
